// react-flexgrid/lib/Row.js
'use strict';

var _extends = require('babel-runtime/helpers/extends')['default'];

var React = require('react');
var classNames = require('classnames');

var ModificatorType = React.PropTypes.oneOf(['xs', 'sm', 'md', 'lg']);
var modificatorKeys = ['start', 'center', 'end', 'top', 'middle', 'bottom', 'around', 'between', 'first', 'last'];

var Row = React.createClass({
  displayName: 'Row',

  propTypes: {
    reverse: React.PropTypes.bool,
    start: ModificatorType,
    center: ModificatorType,
    end: ModificatorType,
    top: ModificatorType,
    middle: ModificatorType,
    bottom: ModificatorType,
    around: ModificatorType,
    between: ModificatorType,
    first: ModificatorType,
    last: ModificatorType
  },
  render: function render() {
    var modificators = ['row'];
    for (var i = 0; i < modificatorKeys.length; ++i) {
      var key = modificatorKeys[i];
      var value = this.props[key];
      if (value) {
        modificators.push('' + key + '-' + value);
      }
    }

    if (this.props.reverse) {
      modificators.push('reverse');
    }

    var className = classNames(this.props.className, modificators);

    return React.createElement(
      'div',
      _extends({}, this.props, { className: className }),
      this.props.children
    );
  }
});

module.exports = Row;
